FreeRTOS/FreeRTOS
# ATmega328PB Xplained Mini -- demo project for ATmega port.

Description
-----------
We previously accepted a third party port for ATmega device family, and the port was verified with Atmel Studio simulator. Since now hardware has arrived, setting up a real demo. (This demo runs on both hardware and integrated simulator.)

As for why this particular board is selected — I'm aware there's only 2KB RAM. I do have a couple Arduino boards at hand. And there's no good out-of-box debug support. Some boards do have debugWIRE interface, ISP interface, ... I don't have proper tool at hand to modify the hardware for debugging purpose. So that picking a board with onboard debug seems to be easier. I also have ATmega4809 and AVR Dx boards, and per discussion those will be supported in separate kernel port(s), due to register address and mapping differences. 

Test Steps
-----------
- Download and install Atmel Studio 7 on Windows. 
- Open project under ```./FreeRTOS/Demo/AVR_ATmega328PB_Xplained_mini_GCC/RTOSDemo``` directory.
- Build project as you would normally do. 
- Program device and debug with debugger mEBDG debugWIRE interface.

Should see onboard LED blink and no test error in check task. 

Reference
-----------
Board information: https://www.microchip.com/DevelopmentTools/ProductDetails/atmega328pb-xmini
ATmega device family: https://www.microchip.com/design-centers/8-bit/avr-mcus (See the PDF.)

Related Issue
-----------
Simulator project: https://github.com/FreeRTOS/FreeRTOS/pull/55
Kernel port PR: https://github.com/FreeRTOS/FreeRTOS-Kernel/pull/48

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
